app/models/user.rb

```diff
@@ -278,6 +278,26 @@ def served_in_military?
     edipi.present? && veteran? || military_person?
   end
 
+  # A user can have many UserPreferences which represent a given PreferenceChoice.
+  # This method will return an array of objects including each Preference and the
+  # associated choices those which the User has opted-in
+  #
+  # @return [Array] an array of objects with paired Preference/UserPreference objects
+  #
+  def preferences
+    results = []
+    user_preferences = UserPreference.eager_load(:preference, :preference_choice).where(account_id: account.id)
+    preference_ids = user_preferences.pluck(:preference_id).uniq
+    preference_ids.each do |id|
+      preference = Preference.find(id)
```

Comment:
Since you eager loaded `Preference` above, I believe you should be able to pull the `Preference` from that `user_preferences` var, b/c I believe this `Preference.find(id)` would still make a brand new db `GET` call for each `Preference`.

Reply:
Kind of gross but here's what I came up with:

```rb
 def preferences
    results = []
    user_preferences = UserPreference.eager_load(:preference, :preference_choice).where(account_id: account.id)
    user_preferences.group_by(&:preference_id).each do |up|
      preference = up[1][0].preference
      results += [{
        preference: preference,
        user_preferences: user_preferences.select { |el| el.preference_id == preference.id }.map(&:preference_choice)
      }]
    end
    results
  end
``` 

Thoughts? The `group_by` clause makes it a bit nested within layers of array, but organizes by `Preference`, taking advantage of the eager loaded records

Reply:
Hey, got a couple ideas.  I'll ping you offline to chat in real time.